snyk-import/snyk-api.ts

```diff
@@ -0,0 +1,114 @@
+// deno-lint-ignore-file no-explicit-any
+
+import { API } from "./const.ts";
+
+type Org = {
+  id: string;
+  name: string;
+  slug: string;
+};
+
+export type Project = {
+  org: Org;
+  id: string;
+  name: string;
+};
+
+export async function getOrgs(token: string): Promise<Org[]> {
+  const orgs = await getSingle(
+    token,
+    `/rest/orgs?version=2023-09-14&limit=100`,
+  );
+
+  if (orgs.next) {
+    throw new Error("More than 100 orgs is not supported");
+  }
+
+  return orgs.data.map((org: any) => ({
+    id: org.id,
+    name: org.attributes.name,
+    slug: org.attributes.slug,
+  }));
+}
+
+export async function getProjects(token: string, org: Org): Promise<Project[]> {
+  const projects: Project[] = [];
+  let next: string | null =
+    `/rest/orgs/${org.id}/projects?version=2023-09-14&limit=10`;
+
+  try {
+    while (next) {
+      const cur = await getSingle(token, next);
```

Comment:
Neither `getSingle` nor `next` are great names here imo. I had to spend some non-insignificant time figuring out what this statement means.

`getSingle` should somehow include that it is meant for sending requests.

`next` should include that it is a request URI.

Reply:
That's fair. Since it's an abstraction around pagination, maybe `getPage()` would be better. As for `next`, I could probably just use `path` or `uri`